docs/src/user-docs/guides-multi-host.md

```diff
@@ -0,0 +1,350 @@
+
+The last step in the previous section (`sbin/start-clp.sh --setup-only`) will generate any necessary
+configuration files, but they're unsuitable for use across multiple hosts (they're designed for use
+on a single host).
+
+:::{note}
+As mentioned at the beginning of this guide, this setup will be made simpler in a future release.
+:::
+
+To update the generated configuration files for use across multiple hosts:
+
+1. **Edit `var/log/.clp-config.yml`**:
+
+    * Update all `host` fields to use the actual hostname or IP address where each service will
+      run (matching what you configured in `etc/clp-config.yml`).
+    * Similarly, update any `port` fields.
+    * For example, if your database runs on `192.168.1.10:3306`, ensure `database.host` is set to
+      `192.168.1.10` and `database.port` is `3306`.
+
+2. **Edit `var/www/webui/server/dist/settings.json`**:
+
+    * Update `SqlDbHost` to the actual hostname or IP address of your database service.
+    * Update `SqlDbPort` if you changed the database port.
+    * Update `MongoDbHost` to the actual hostname or IP address of your results cache service.
+    * Update `MongoDbPort` if you changed the results cache port.
+
+### Distributing the set-up package
+
+With the package set up, we can now distribute it to all hosts in the cluster:
+
+1. Copy the set-up package to all hosts where you want to run CLP services.
+
+    * Ensure the package is copied to the same location on every host or else, on each host, you'll
+      need to modify the paths in `.env` as appropriate.
+
+2. Configure worker concurrency (optional):
+
+   On each worker host, edit the `.env` file to adjust worker concurrency settings as needed:
+
+   * `CLP_COMPRESSION_WORKER_CONCURRENCY`
+   * `CLP_QUERY_WORKER_CONCURRENCY`
+   * `CLP_REDUCER_CONCURRENCY`
+
+   Recommended settings:
+
+   * If workers are started on separate hosts, set each concurrency value to match the CPU count on
+     that host.
+   * If compression and query/reducer workers are started on the same host, set each concurrency
+     value to half the CPU count (e.g., for a 16-core host, set all three to 8).
+
+## Starting CLP
+
+You can start CLP across multiple hosts by starting each service on the relevant host. The commands
+below indicate how to do so, with comments indicating the startup order and dependencies between
+services.
+
+:::{note}
+For **clp-json + Presto** deployments (storage engine: `clp-s` with query engine: `presto`), you
+can omit starting the `query-scheduler`, `query-worker`, and `reducer` services.
+:::
+
+:::{tip}
+If you want to use your own MariaDB/MySQL or MongoDB servers instead of the Docker Compose managed
+databases, see the [external database setup guide](guides-external-database.md). When using external
+databases, skip starting the `database` and `results-cache` services below.
+:::
+
+All commands below assume you are running them from the root of the CLP package directory.
+
+```bash
```

Comment:
1. with the latest code, i believe the `--file docker-compose-all.yaml` argument should be removed
2. some services use `--wait` while others use `--detach`. i believe that's to speed up the launches? if so, shall we add a note / tip / inline comment to warn users about the differences are intentional? (at least they should understand that `--detach` can't be blindly used)

Reply:
Sorry, missed this notification. Yeah, let's consistently use `--wait` if it works.